reco_utils/dataset/covid_utils.py

```diff
@@ -0,0 +1,201 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+from reco_utils.dataset import download_utils
+from io import StringIO
+import pandas as pd
+import numpy as np
+import json
+
+
+def load_pandas_df(
+    azure_storage_account_name='azureopendatastorage',
+    azure_storage_sas_token='sv=2019-02-02&ss=bfqt&srt=sco&sp=rlcup&se=2025-04-14T00:21:16Z&st=2020-04-13T16:21:16Z&spr=https&sig=JgwLYbdGruHxRYTpr5dxfJqobKbhGap8WUtKFadcivQ%3D',
+    container_name='covid19temp',
+    metadata_filename='metadata.csv'
+    ):
+    """ Loads the Azure Open Research COVID-19 dataset as a pd.DataFrame.
+
+    The Azure COVID-19 Open Research Dataset may be found at https://azure.microsoft.com/en-us/services/open-datasets/catalog/covid-19-open-research/
+
+    Args:
+        azure_storage_account_name (str): Azure storage account name.
+        azure_storage_sas_token (str): Azure storage SaS token.
+        container_name (str): Azure storage container name.
+        metadata_filename (str): Name of file containing top-level metadata for the dataset.
+    
+    Returns:
+        metadata (pd.DataFrame): Metadata dataframe.
+        blob_service (azure.storage.blob.BlockBlobService): Azure BlockBlobService for dataset.
+    """
+
+    # Get metadata (may take around 1-2 min)
+    blob_service = download_utils.get_blob_service(azure_storage_account_name, azure_storage_sas_token, container_name)
+    metadata = download_utils.load_csv_from_blob(blob_service, container_name, metadata_filename)
+
+    return metadata, blob_service
+
+def extract_public_domain(metadata):
```

Comment:
While this could easily be done in the notebook, I feel this wrapper function makes the data preprocessing section seem simpler in the notebook.

@miguelgfierro @anargyri What are your thoughts?

Reply:
This has now been removed.